tests/org.pitest.pitclipse.ui.tests/src/org/pitest/pitclipse/ui/behaviours/pageobjects/PitSummaryView.java

```diff
@@ -16,34 +16,87 @@
 
 package org.pitest.pitclipse.ui.behaviours.pageobjects;
 
-import org.pitest.pitclipse.ui.behaviours.StepException;
-import org.pitest.pitclipse.ui.swtbot.PitNotifier;
-import org.pitest.pitclipse.ui.swtbot.PitResultsView;
+import java.util.concurrent.atomic.AtomicReference;
 
-public class PitSummaryView {
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swtbot.eclipse.finder.SWTWorkbenchBot;
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotBrowser;
+import org.pitest.pitclipse.ui.view.PitView;
 
-    private PitResultsView lastResults = null;
+public class PitSummaryView {
+    private PitView summaryView;
+    private SWTWorkbenchBot bot;
+    private SWTBotBrowser browser;
 
-    public PitSummaryView() {
+    public PitSummaryView(SWTWorkbenchBot bot) {
+        this.bot = bot;
     }
 
-    public void waitForUpdate() {
-        try {
-            lastResults = PitNotifier.INSTANCE.getResults();
-        } catch (InterruptedException e) {
-            throw new StepException(e);
+    public void getViewIfNotSet() {
+        if (summaryView == null) {
+            summaryView = (PitView) Views.getViewById(PitView.VIEW_ID);
```

Comment:
Could you add a try/catch that logs an error in case the cast to `PitView` fails?

Reply:
Sure